#!/usr/bin/env python
PACKAGE = 'goal_manager'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('publish_goals',             bool_t,     0, '', True)
gen.add('read_mission_file',        bool_t,     0, '', False)
gen.add('filename',              str_t,      0, 'Fixed Frame', 'test.txt')
gen.add('workspace',              str_t,      0, 'Fixed Frame', '/home/talos/rosworkspace/src/talos/waypoints')
gen.add('wait_for_userstart',      bool_t,     0, '', True)
gen.add('wp_approximation',     double_t,   0, 'approximate distance to waypoint required in [m]', 0.2, 0, 10.)
gen.add('path_frequency',     double_t,   0, 'publish frequency of path', 0.1, 10, 100.)

gen.add('waittime',   double_t,   0, 'Timeout at reached goalpoint [s]', 1.0, 0, 5)
gen.add('fixed_frame',              str_t,      0, 'Fixed Frame', 'odom')
gen.add('rabbit_frame',              str_t,      0, 'Fixed Frame', 'rabbit')
gen.add('robot_frame',              str_t,      0, 'Fixed Frame', 'base_link')

#gen.add('kalman_p',                 double_t,   0, '',    4, 0, 10)
#gen.add('kalman_q',                 double_t,   0, '', .002, 0, 10)
#gen.add('kalman_r',                 double_t,   0, '',   10, 0, 20)
#gen.add('kalman_on',                int_t,      0, '',    1, 0,  1)

exit(gen.generate(PACKAGE, 'goal_manager', 'GoalManager'))
